### PR TITLE
[FIX] Remove references to unspecified aslcontext.json

### LINKED ISSUE #1479: [BUG] *_aslcontext.json should probably not be in the spec
### Describe your problem in detail.

brought to my attention by @patsycle 

The original [ASL BEP](https://github.com/bids-standard/bids-specification/pull/669/files) did not mention any filename pattern that could end with *_aslcontext.json.

See here:

https://github.com/bids-standard/bids-specification/blob/b92b6eaeab19fb4276e489dc04e3cae523f82aa4/src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md#arterial-spin-labeling-perfusion-data

However there seemed to have been a small inconsistency in the BEP as the schema for ASL did support both `*_aslcontext.json` and `*_aslcontext.tsv` (the latter being the only one mentioned / allowed in the BEP).

https://github.com/bids-standard/bids-specification/pull/669/files#r1180810465

When we switched to using the schema as "ground truth", the rendering of the filename patterns with the macro started displaying both files.

The fix should be easy and just removing one line from the schema.

I checked and the latest version of the validator does not allow this type of file:

```
bids-validator asl001
bids-validator@1.11.1-dev.0
        1: [ERR] Files with such naming scheme are not part of BIDS specification. This error is most commonly caused by typos in file names that make them not BIDS compatible. Please consult the specification and make sure your files are named correctly. If this is not a file naming issue (for example when including files not yet covered by the BIDS specification) you should include a ".bidsignore" file in your dataset (see https://github.com/bids-standard/bids-validator#bidsignore for details). Please note that derived (processed) data should be placed in /derivatives folder and source data (such as DICOMS or behavioural logs in proprietary formats) should be placed in the /sourcedata folder. (code: 1 - NOT_INCLUDED)
                ./sub-Sub103/perf/sub-Sub103_aslcontext.json
                        Evidence: sub-Sub103_aslcontext.json

        Please visit https://neurostars.org/search?q=NOT_INCLUDED for existing conversations about this issue.

        Please visit https://neurostars.org/search?q=NIFTI_TOO_SMALL for existing conversations about this issue.

        Summary:                Available Tasks:        Available Modalities: 
        9 Files, 95.59KB                                MRI                   
        1 - Subject                                                           
        1 - Session            
```

So I don't think we will break any dataset.

### Describe what you expected.

no allowed filename pattern for `*_aslcontext.json`

### BIDS specification section

https://bids-specification.readthedocs.io/en/latest/modality-specific-files/magnetic-resonance-imaging-data.html#arterial-spin-labeling-perfusion-data

### PR DESCRIPTION
**Pull request by Remi-Gau**
fixes #1479

- remove aslcontext.json
- add png and tif extensions for asllabelling
- add paragraph describing asllabelling images

### CODE REVIEW

**Review by tsalo — COMMENTED** [review]
Everything looks good to me, except for the wording of the paragraph about the asllabeling file.

File: src/modality-specific-files/magnetic-resonance-imaging-data.md
Comment (Remi-Gau) [reply src/modality-specific-files/magnetic-resonance-imaging-data.md:911]: Given i took this from the glossary it may be good to edit it there too for internal consistency.

**Review by tsalo — APPROVED** [review]
LGTM!

File: src/modality-specific-files/magnetic-resonance-imaging-data.md
Comment (effigies) [inline src/modality-specific-files/magnetic-resonance-imaging-data.md:913]:
```suggestion
See [`LabelingLocationDescription`](glossary.md#labelinglocationdescription-metadata) for more details.
```

**Review by effigies — APPROVED** [review]
One small suggestion and an additional suggestion for a follow-up PR.

**Comment by Remi-Gau** [comment]
I would suggest we wait for some ASL BEP lead to also approve this before merging.

**Comment by effigies** [comment]
@patsycle @HenkMutsaerts In case you missed the review request...

**Review by HenkMutsaerts — APPROVED** [review]

File: src/modality-specific-files/magnetic-resonance-imaging-data.md
Comment (Remi-Gau) [reply src/modality-specific-files/magnetic-resonance-imaging-data.md:911]: you mean creating a new macro then, right?

Comment (Remi-Gau) [reply src/modality-specific-files/magnetic-resonance-imaging-data.md:911]: I believe this relates to this:
https://github.com/bids-standard/bids-specification/issues/1038

**Pull request merged**
